neo/Core/TX/Transaction.py

```diff
@@ -774,3 +774,19 @@ def __init__(self, *args, **kwargs):
         """
         super(ContractTransaction, self).__init__(*args, **kwargs)
         self.Type = TransactionType.ContractTransaction
+
+
+class TransactionError(Exception):
+    """
+    Provide user-friendly feedback for specific transaction exceptions.
+    """
+
+    message = None
+
+    def __init__(self, message):
+        super(TransactionError, self).__init__(message)
+        self.message = message
+
+    @staticmethod
+    def FeeError(message=None):
```

Comment:
I understand that you copied the structure used for the JSON server. For this case though I don't think it's optimal. The JSON server is very specific in which it needs an error code and message. More important is that it is also converted to the JSON/text format. Meaning something like so:
```
"error": {"code": -32700, "message": "Parse error"}}
```
The only way the receiving party can differentiate is on `error.message` or `message` alone. They _have_ to do string comparisons to decide how to act. 

In our case the `wallet.MakeTransaction` function is always called from other python code and so can benefit from handling python specific exceptions without having to parse `exception.message`. For example, say we extend your `TransactionError` to also include a `InsufficientFunds` static method. We'd have to do something like:
```
try:
    tx = wallet.MakeTransaction
except TransactionError as te:
   if "insufficient funds" in te.message:
       #do something
   elif "fee" in te.message:
      #do something
   else:
      # default do something
```

if we would make exceptions specific like e.g.
```
class TXFundsError(Exception):
   pass

class TXFeeError(Exception):
   pass
``` 
We can then do
```
try:
   tx = wallet.Maketransaction
except TXFundsError as e:
   #do something if insufficient funds
except TXFeeError as e:
  # do something if insufficient fees
```
This is more pythonic, easier to control (little bit of background [info](https://docs.python.org/3/tutorial/errors.html#user-defined-exceptions)) and increases re-usability of `wallet.MakeTransaction`. Note that big frameworks like Django do this [as well]( 
https://docs.djangoproject.com/en/2.1/_modules/django/core/exceptions/)

Reply:
@ixje Thank you for the in depth explanation!